### ISSUE TITLE
Migrating to TimescaleDB failed on inserting duplicates into idata from idata_xxxx 

### ISSUE THREAD

**Issue by CapNem0** [netxms/netxms#83]
Migration to TimescaleDB failed,   if a source idata_XXX table contain multiple records with same item_id and data_timestamp:

```
Migrating table idata_287 to data
ERROR: unable to insert data to destination table (23505 ERROR:  duplicate key value violates unique constraint "349_349_idata_sc_30_pkey"
DETAIL:  Key (item_id, idata_timestamp)=(1642, 2023-03-04 13:54:13+03) already exists.)
ERROR: unable to insert data to destination table (23505 ERROR:  duplicate key value violates unique constraint "349_349_idata_sc_30_pkey"
DETAIL:  Key (item_id, idata_timestamp)=(1642, 2023-03-04 13:39:13+03) already exists.)
ERROR: unable to insert data to destination table (23505 ERROR:  duplicate key value violates unique constraint "349_349_idata_sc_30_pkey"
DETAIL:  Key (item_id, idata_timestamp)=(1641, 2023-03-04 12:51:11+03) already exists.)
ERROR: unable to insert data to destination table (23505 ERROR:  duplicate key value violates unique constraint "349_349_idata_sc_30_pkey"
DETAIL:  Key (item_id, idata_timestamp)=(1641, 2023-03-04 12:50:11+03) already exists.)
ERROR: unable to insert data to destination table (23505 ERROR:  duplicate key value violates unique constraint "349_349_idata_sc_30_pkey"
DETAIL:  Key (item_id, idata_timestamp)=(1641, 2023-03-04 13:23:13+03) already exists.)
ERROR: unable to insert data to destination table (23505 ERROR:  duplicate key value violates unique constraint "349_349_idata_sc_30_pkey"
DETAIL:  Key (item_id, idata_timestamp)=(1642, 2023-03-04 13:24:13+03) already exists.)
ERROR: unable to insert data to destination table (23505 ERROR:  duplicate key value violates unique constraint "349_349_idata_sc_30_pkey"
DETAIL:  Key (item_id, idata_timestamp)=(1641, 2023-03-04 13:24:13+03) already exists.)
```
In my situation duplicate records can contain the same values or different values. 

Probably, function  MigrateSingleDataTableToTSDB [https://github.com/netxms/netxms/blob/master/src/server/tools/nxdbmgr/migrate.cpp](https://github.com/netxms/netxms/blob/master/src/server/tools/nxdbmgr/migrate.cpp) should check target table for existance of records with this item_id and idata_timestamp (unique key pair) or skip migration of failed record and continue with next one.

Using DISTINCT in a select clause con solve situation if the same values in duplicates . 
 
With some modifications to migrate.cpp got working migration (patch attached ) Of course this is not best solution, but solved my problem. Failed records skipped and import continues. 

First migration  MariaDB -> PostgreSQL - saved about 20G 
Second Migration PostgreSQL -> TimescaleDB  in progress

Ubuntu 18.04 NetXMS 4.3.2 MariaDB 10.1.4 
[migrate.cpp.patch](https://github.com/netxms/netxms/files/10892576/migrate.cpp.patch)
![Снимок экрана 2023-03-05 в 20 41 05](https://user-images.githubusercontent.com/1137679/222978317-d93259ff-9952-4022-9eb0-9e8d049fa177.png)

**Comment by CapNem0** [comment]
Some corrections in a patch was made after migrationg: if migrating from idata_XXX has error state - migrating from tdata_XXX silently skipped 
[migrate.cpp.patch](https://github.com/netxms/netxms/files/10894908/migrate.cpp.patch)

**Comment by CapNem0** [comment]
Some corrections made to previos patch - if some records was skipped from idata_xxx, then also skipped all migration from tdata_XXX. Corrected.

Hope this patch can help migrating MariaDB -> TimescaleDB  with strange duplicates in a source DB

[migrate.cpp.patch](https://github.com/netxms/netxms/files/10895241/migrate.cpp.patch)

**Comment by 2128506** [comment]
Fixed by commits 1a4f0dd and b8f2f52. Instead of doing SELECT DISTINCT (which can significantly decrease performance on large tables) I've added ON CONFLICT DO NOTHING to inserts into TSDB. Also added new nxdbmgr command line option -x that instructs to continue migration even if there are collected data migration errors.
